### ISSUE TITLE
.getCurrentAccount() show error

### ISSUE THREAD

**Issue by creativefull** [underscopeio/react-native-facebook-account-kit#36]
hi, i'm trying like a tutorial but i got error like this

```
Possible Unhandled Promise Rejection (id: 0):
Server generated an error
Error: Server generated an error
    at createErrorFromErrorData (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7452:11)
    at http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7403:11
    at MessageQueue.__invokeCallback (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7266:10)
    at http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7138:8
    at guard (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7076:1)
    at MessageQueue.invokeCallbackAndReturnFlushedQueue (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=false&minify=false:7137:1)
    at DedicatedWorkerGlobalScope.onmessage (http://localhost:8081/debuggerWorker.js:39:56)
```

my code

```
	componentWillMount() {
		AccountKit.configure({
			defaultCountry : 'ID'
		});
		AccountKit.getCurrentAccessToken().then(
			(token) => {
				if (!token) {
					AppRegistry.runApplication('login', {
						initialProps : {},
						rootTag : 1
					});
				}
				else {
					AccountKit.getCurrentAccount().then(
						(account) => {
							console.log("Current Account ", account);
						}).catch(function(error) {
							console.log('There has been a problem with your fetch operation: ' + error.message);
							// ADD THIS THROW error
							throw error;
						});
				}
		}).catch((error) => {
			console.log(error.message);
			throw error;
		});
	}
```

is there anything I missed in the code section?
please help me :(

**Comment by gaguirre** [comment]
Are you sure that error is thrown by `getCurrentAccount` method?
Could you try isolating the code?

Thanks!

**Comment by creativefull** [comment]
what it means isolating code?

sorry i'm newbie

**Comment by gaguirre** [comment]
I meant removing all code that isn't related to Account Kit specifically.
Have you tried running the sample?

**Comment by creativefull** [comment]
sample code or sample app ?

**Comment by gaguirre** [comment]
This project's sample. You can follow [this steps](https://github.com/underscopeio/react-native-facebook-account-kit#examples).

**Comment by creativefull** [comment]
yes, i've been follow that steps, but still not working for me ,.

my phone using lollipop,

**Comment by creativefull** [comment]
successful for login phone,. but after login i did'nt got account :(

**Comment by arhmnsh** [comment]
+1 
Facing the exact same issue on Android. Works fine on iOS.
Sample app works fine. So, removed all my custom code, and copy-pasted code from the sample. Still getting this error on getCurrentAccount().

**Comment by gaguirre** [comment]
Which version of React Native are you using guys? @creativefull @arhmnsh 
Did you [configure the project](https://github.com/underscopeio/react-native-facebook-account-kit#configuration) with your credentials?

**Comment by arhmnsh** [comment]
RN is 0.32. And yep, I configured it properly.

**Comment by creativefull** [comment]
+1
waiting for fix in android

**Comment by gaguirre** [comment]
Ok, thank you guys, I'll debug it later.
It's probably related to the RN version.

**Comment by arhmnsh** [comment]
Hi @gaguirre 
Is there any update on this?

**Comment by gaguirre** [comment]
I just published a new release (**0.5.0**) compatible with **React Native 0.33**.
Would you try it and see if it fixes the error?

Notice that, from now on, there are two separated examples for **RN 0.24** and **RN 0.33**.

**Comment by creativefull** [comment]
ok i'll try again ,. thanks

**Comment by arhmnsh** [comment]
Hey @gaguirre,
Thanks for the update buddy.
That didn't work. But I think I found the issue. I went the though the sample app code, found some differences in `strings.xml` and `AppManifest.xml`. 

*strings.xml:*
<img width="569" alt="screen shot 2016-09-20 at 2 53 54 pm" src="https://cloud.githubusercontent.com/assets/1363724/18669152/3e279114-7f42-11e6-9e62-9577b69bc262.png">

*AndroidManifest.xml:*
<img width="564" alt="screen shot 2016-09-20 at 2 54 17 pm" src="https://cloud.githubusercontent.com/assets/1363724/18669164/4cb6117e-7f42-11e6-9e5f-951230eea7cb.png">




So I took the changes from the sample, applied it in my project and it worked.

Hope it helps.

**Comment by arhmnsh** [comment]
Oops nope! It just worked for a couple of times. Facing the same issue again now 😕

**Comment by gaguirre** [comment]
I can't reproduce the error. 
Try with the [example](https://github.com/underscopeio/react-native-facebook-account-kit#examples). If it works fine then the error must be something in your project.

**Comment by arhmnsh** [comment]
@gaguirre Example works perfectly. Just tried to create a new 0.33-based project, worked fine for a couple of runs, and then the same issue. :(

**Comment by gaguirre** [comment]
@arhmnsh could you post the error stacktrace? or share a repo with the project?

**Comment by arhmnsh** [comment]
@gaguirre I created this project just with accountKit. Here it is: https://gitlab.com/arhmnsh/fbakTest/blob/master/README.md

**Comment by arhmnsh** [comment]
Hi @gaguirre, you had any chance of checking out the repo?

**Comment by gaguirre** [comment]
I'm looking at it right now.
A new version of Android SDK was released these days so I'll upgrade it and let you know.

**Comment by gaguirre** [comment]
@arhmnsh with a deeper debug I can tell your error is related to a configuration on the FB admin page.
Try with [the answer to this stackoverflow question](http://stackoverflow.com/a/22909841/3791502) and tell me if that works :)

I'll improve the error handling to show better error messages.

**Comment by Andreyco** [comment]
@gaguirre while the advice on SO works, it deserves more than hotfix. Is there any plan, how to properly solve this issue?
TY

**Comment by gaguirre** [comment]
I'm not sure 100% but I think sending the `app_proof` is not supported by the SDK.
However, that could have changed or I could be wrong, so if anyone find another better solution would be great!

**Comment by Andreyco** [comment]
As I recall, there is no newer SDK version. I suppose, we should wait then...

**Comment by gaguirre** [comment]
Good idea @Andreyco. Feel free to send a PR for this too ;)
Thanks for contributing!!

**Comment by creativefull** [comment]
Thanks @Andreyco it is work fine :+1:

**Comment by edo1493** [comment]
Hello, I am on the latest version and RN 0.48.4. I am logging in through the phone number and I get back my token. However, when I try to retrieve the account or access token, I get an empty object in the first case and undefined.

**Comment by edo1493** [comment]
@gaguirre I have also tried to call configure, before trying to get the account object, but it doesn't make any difference.

**Comment by creativefull** [comment]
hi @edo1493 u need to setting in developer.facebook.com to disable secret key. because if u use it maybe not working on android :)

**Comment by creativefull** [comment]
![image](https://user-images.githubusercontent.com/13174684/31364829-a471a6d4-ad91-11e7-87df-d62eb82359d6.png)

try make your setting on product account kit like this :)

**Comment by azharkiz** [comment]
i am using Facebook Account Kit for OTP verification in react native. everything is working is working but i can't find a way to get the mobile number which is verified.. please help me guys?

   RNAccountKit.getCurrentAccount().then(account => {
       console.log(`Current account: ${account}`)
      })
    }

**Comment by amanix007** [comment]
![errorapture](https://user-images.githubusercontent.com/13711025/40276922-97f73e32-5c37-11e8-996a-b2affa4f5b31.PNG)
Even my "Require App Secret" set to "No" and "Enable Client Access Token Flow" set to "Yes" Still Having same issue.

**Comment by azharkiz** [comment]
In my case i wrote Require App Secret" set to "yes" then i am changed it
into "no" at that time i got the account details. In your case i suggestion
you check your code.

**Comment by rassemdev** [comment]
@azharkiz can you share your code? i'm new in react-native

**Comment by azharkiz** [comment]
You refer the facebook account kit authentication react native youtube 1 hr
video you can get the result..

**Comment by hutsi** [comment]
having the same issue, setting "Require App Secret" to "NO" didn't do the trick..

**Comment by creativefull** [comment]
anyone got a solution for this error ?

**Comment by lilian131** [comment]
@creativefull hi I had the same problem with a cordova plugin in ionic 
for me it was because despite removing the token with a normal way when logout
```
localStorage.removeItem('token');
```
I had to use the method provided to destroy the token like 
```
AccountKitPlugin.logout();
```
and it works!!
Hope this will help other because this was something really frustrating as it made my server always crashed
